src/smt/set_defaults.cpp

```diff
@@ -46,6 +46,56 @@ using namespace cvc5::internal::theory;
 namespace cvc5::internal {
 namespace smt {
 
+/**
+ * Set domain.optName to value due to reason. Notify if value changes.
+ */
+#define SET_AND_NOTIFY(domain, optName, value, reason) \
+  if (opts.write##domain().optName != value)           \
+  {                                                    \
+    notifyModifyOption(#optName, #value, reason);      \
+    opts.write##domain().optName = value;              \
+  }
+/**
+ * Set domain.optName to value due to reason. Notify if value changes.
+ * Print the value using a stringstream conversion.
+ *
+ * Note this macro should be used if the value passed to the macro is not
+ * concrete (e.g. is stored in a variable).
+ */
+#define SET_AND_NOTIFY_VAL_SYM(domain, optName, value, reason) \
+  if (opts.write##domain().optName != value)                   \
+  {                                                            \
+    std::stringstream sstmp;                                   \
+    sstmp << value;                                            \
+    notifyModifyOption(#optName, sstmp.str(), reason);         \
+    opts.write##domain().optName = value;                      \
+  }
+/**
+ * Set domain.optName to value due to reason if the option was not already set
+ * by the user. Notify if value changes.
+ */
+#define SET_AND_NOTIFY_IF_NOT_USER(domain, optName, value, reason) \
+  if (!opts.write##domain().optName##WasSetByUser                  \
+      && opts.write##domain().optName != value)                    \
+  {                                                                \
+    notifyModifyOption(#optName, #value, reason);                  \
+    opts.write##domain().optName = value;                          \
+  }
+/**
+ * Set domain.optName to value due to reason if the option was not already set
+ * by the user. Notify if value changes. Print the value using a stringstream
+ * conversion.
```

Comment:
I think the last sentence can be removed. Add a note similarly to the one for `SET_AND_NOTIFY_VAL_SYM`.